src/params.rs

```diff
@@ -185,6 +185,199 @@ pub struct ListPaginator<T, P> {
     pub params: P,
 }
 
+#[derive(Debug)]
+pub struct SearchListPaginator<T, P> {
```

Comment:
I would rather we make ListPaginator be generic over some trait called `PaginableList` which SearchList and List both implement. You can use GATs to have that trait have some generic associated type (the T in this case), that way we don't have duplicated logic for both types of paginator.

Reply:
Hi thanks for the review. I have implemented the suggested changes. I am weak with advanced traits and GAT's so I am not sure if I have implemented it correctly (although I am %99 sure it's fine). Feel free to review it and let me know if I should change anything. Cheers!